lib/sinon.js

```diff
@@ -7,7 +7,21 @@
  * Copyright (c) 2010-2013 Christian Johansen
  */
 "use strict";
-module.exports = require("./sinon/util/core");
+
+var match = require("./sinon/match");
+
+module.exports = exports = require("./sinon/util/core");
+
+exports.extend = require("./sinon/extend");
+exports.match = match;
+exports.typeOf = require("./sinon/typeOf");
+
+/*
+ * allow deepEqual to check equality of matchers through
+ * dependency injectection. Otherwise we get a circular
+ * dependency
+ */
+exports.deepEqual.match = match;
```

Comment:
On the road to the solution you see here, I had `require('.../core/deep-equal').withMatcher(match)` in my code at some point. I removed it in favor of this solution which seemed cleaner at the time.

I also tried:

* `deepEqual.upgrade(m)`
* `deepEqual.allowMatcher(m)`
* `deepEqual.checkMatcher(m)`

Just throwing out ideas. `.use(match)` for some reason sounds better to me than `.using`. I'm flexible though.